t/rest_server/main_test.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/abcxyz/pkg/logging"
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestRealMain(t *testing.T) {
+	ctx := logging.WithLogger(context.Background(), logging.TestLogger(t))
+	ctx, done := context.WithCancel(ctx)
+	defer done()
+
+	errCh := make(chan error, 1)
+	doneCh := make(chan struct{}, 1)
+	go func() {
+		defer close(doneCh)
+
+		if err := realMain(ctx); err != nil {
+			select {
+			case errCh <- err:
+			default:
+			}
+		}
+	}()
+
+	cmd := exec.Command(
+		"curl",
```

Comment:
Not sure we should take a dependency on curl. My suggestion would be:

 - Here in this function, use `http.Get()` instead of exec'ing curl, to make a real network request
 - And also create a separate test that uses httptest (or whatever the convention is for abcxyz, I don't recall) to just test the handlers without a listening on a real port

Reply:
Changed to http.get(), Added separate test for handle function.